vsintegration/src/FSharp.Editor/CodeFix/RemoveUnusedBinding.fs

```diff
@@ -0,0 +1,98 @@
+
+namespace Microsoft.VisualStudio.FSharp.Editor
+
+open System
+open System.Composition
+open System.Threading.Tasks
+
+open Microsoft.CodeAnalysis.Text
+open Microsoft.CodeAnalysis.CodeFixes
+
+open FSharp.Compiler.CodeAnalysis
+open FSharp.Compiler.Syntax
+open FSharp.Compiler.Text
+
+[<AutoOpen>]
+module FSharpParseFileResultsExtensions =
+    type FSharpParseFileResults with
+        member this.TryRangeOfBindingWithHeadPatternWithPos pos =
+            let input = this.ParseTree
+            SyntaxTraversal.Traverse(pos, input, { new SyntaxVisitorBase<_>() with 
+                member _.VisitExpr(_, _, defaultTraverse, expr) =
+                    defaultTraverse expr
+
+                override _.VisitBinding(_path, defaultTraverse, binding) =
+                    match binding with
+                    | SynBinding(_, SynBindingKind.Normal, _, _, _, _, _, pat, _, _, _, _) as binding ->
+                        if Position.posEq binding.RangeOfHeadPattern.Start pos then
+                            Some binding.RangeOfBindingWithRhs
+                        else
+                            // Check if it's an operator
+                            match pat with
+                            | SynPat.LongIdent(LongIdentWithDots([id], _), _, _, _, _, _) when id.idText.StartsWith("op_") ->
+                                if Position.posEq id.idRange.Start pos then
+                                    Some binding.RangeOfBindingWithRhs
+                                else
+                                    defaultTraverse binding
+                            | _ -> defaultTraverse binding
+
+                    | _ -> defaultTraverse binding })
+
+[<ExportCodeFixProvider(FSharpConstants.FSharpLanguageName, Name = "RemoveUnusedBinding"); Shared>]
+type internal FSharpRemoveUnusedBindingCodeFixProvider
+    [<ImportingConstructor>]
+    (
+        checkerProvider: FSharpCheckerProvider, 
+        projectInfoManager: FSharpProjectOptionsManager
+    ) =
+    
+    inherit CodeFixProvider()
+    static let userOpName = "RemoveUnusedBinding"
+    let fixableDiagnosticIds = set ["FS1182"]
+    let checker = checkerProvider.Checker
+
+    override _.FixableDiagnosticIds = Seq.toImmutableArray fixableDiagnosticIds
+
+    override _.RegisterCodeFixesAsync context : Task =
+        asyncMaybe {
+            // Don't show code fixes for unused values, even if they are compiler-generated.
+            do! Option.guard context.Document.FSharpOptions.CodeFixes.UnusedDeclarations
+
+            let document = context.Document
+            let! sourceText = document.GetTextAsync(context.CancellationToken)
+
+            let! parsingOptions, _ = projectInfoManager.TryGetOptionsForEditingDocumentOrProject(document, context.CancellationToken, userOpName)
+            let! parseResults = checker.ParseFile(document.FilePath, sourceText.ToFSharpSourceText(), parsingOptions, userOpName = userOpName) |> liftAsync
```

Comment:
I recommend to use `ParseDocument` instead of `ParseFile`.

`ParseFile` is fine, but lately we've been consolidating parsing and checking documents through extensions to the `FSharpChecker`, which `ParseDocument` is one of those extensions.